tests/scenarios/v2-addon-dev-watch-test.ts

```diff
@@ -134,6 +137,139 @@ Scenarios.fromProject(() => baseV2Addon())
+            filePath: distPath,
+            assert,
+            // Update a component
+            fn: async () => {
+              let someContent = await fs.readFile(demoHbs);
+
+              // generally it's bad to introduce time dependencies to a test, but we need to wait long enough
+              // to guess for how long it'll take for the file system to update our file.
+              //
+              // the `stat` is measured in `ms`, so it's still pretty fast
+              await fs.writeFile(demoHbs, someContent + `\n\n`);
+              await aBit(10);
+              await watcher?.nextBuild();
+            },
+          });
+
+          // updating hbs content should update resulting app re-exported component
+          distPath = path.join(addon.dir, 'dist/_app_/components/test.js');
+          await isNotModified({
+            filePath: distPath,
+            assert,
+            // Update a component
+            fn: async () => {
+              let someContent = await fs.readFile(demoHbs);
+
+              // generally it's bad to introduce time dependencies to a test, but we need to wait long enough
+              // to guess for how long it'll take for the file system to update our file.
+              //
+              // the `stat` is measured in `ms`, so it's still pretty fast
+              await fs.writeFile(demoHbs, someContent + `\n`);
+              await aBit(10);
+              await watcher?.nextBuild();
+            },
+          });
+
+          // updating template only hbs should update the dist output
+          distPath = path.join(addon.dir, 'dist/components/button.js');
+          await isNotModified({
+            filePath: distPath,
+            assert,
+            // Update a component
+            fn: async () => {
+              let someContent = await fs.readFile(demoHbs);
+
+              // generally it's bad to introduce time dependencies to a test, but we need to wait long enough
+              // to guess for how long it'll take for the file system to update our file.
+              //
+              // the `stat` is measured in `ms`, so it's still pretty fast
+              await fs.writeFile(demoHbs, someContent + `\n`);
+              await aBit(10);
+              await watcher?.nextBuild();
+            },
+          });
+
+          // deleting demo.js should make demo a template only component
+          const demoJsContent = await fs.readFile(demoJs);
+          await fs.rm(demoJs);
+          await watcher?.nextBuild();
+          let distPathDemoCompContent = await fs.readFile(distPathDemoComp);
+          assert.ok(distPathDemoCompContent.includes('templateOnly'));
+
+          // creating demo.js should make demo a template colocated component
+          await fs.writeFile(demoJs, demoJsContent);
+          await watcher?.nextBuild();
+          distPathDemoCompContent = await fs.readFile(distPathDemoComp);
+          assert.ok(!distPathDemoCompContent.includes('templateOnly'));
```

Comment:
nit pick:

```suggestion
          assert.false(distPathDemoCompContent.includes('templateOnly'));
```

(this was initially confusing to me, as it took me a while to see the `!` 😅 )